### DISCUSSION TITLE
How can I set `VCPKG_INSTALLED_RIR` in CMake

### DISCUSSION THREAD

**Discussion by GasimGasimzada** [microsoft/vcpkg#35850] [General]
I have an existing project in C++ that I am trying to port to CMake. I am already using vcpkg for managing all packages and I install all packages to `<repo>/vcpkg_installed` as a separate step. I run it once both during local development and in CI/CD pipeline, then I build my project. Now, when I set up my project to work with CMake (`cmake . -B build`), I get a duplicate `vcpkg_installed` located inside the `build` directory. I want to change it so that my separate vcpkg installation step and cmake use the same `vcpkg_installed` directory. This is especially useful if I want to recreate the build directory but do not want to rebuild all my Vcpkg dependencies again (even though they are cached, they still take time).

**Comment by Neumann-A** [comment]
Pass to CMake: `-DVCPKG_INSTALLED_DIR=<your pre-installed vcpkg path>` ?

**Reply by GasimGasimzada** [reply]
I was trying to figure out how to set it in CMakeLists.txt file. I ended up doing this, which worked:

```cmake
set(VCPKG_INSTALLED_DIR "${PROJECT_SOURCE_DIR}/vcpkg_installed")
```

Thank you for the help!
